src/components/DetaledItem/index.tsx

```diff
@@ -0,0 +1,130 @@
+import { useEffect, useState } from 'react';
+import { fetchAbilityDetail } from '../../api/pokeApi';
+import type { AbilityDetail } from '../../api/constants.ts';
+
+type DetailedItemProps = {
+  id: string | number;
+  onClose: () => void;
+};
+
+export default function DetailedItem({ id, onClose }: DetailedItemProps) {
+  const [data, setData] = useState<AbilityDetail | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
```

Comment:
```suggestion
  const [isLoading, setIsLoading] = useState(false);
```
указание <boolean> тут избыточно